frontend/src/lib/delete-cluster.ts

```diff
@@ -38,11 +40,11 @@ export function deleteCluster(cluster: Cluster, ignoreClusterDeploymentNotFound
     return {
         promise: new Promise((resolve, reject) => {
             deleteResourcesResult.promise.then((promisesSettledResult) => {
-                if (promisesSettledResult[0].status === 'rejected') {
+                if (promisesSettledResult[0]?.status === 'rejected') {
                     reject(promisesSettledResult[0].reason)
                     return
                 }
-                if (promisesSettledResult[1].status === 'rejected') {
```

Comment:
we need to switch around the logic as it expects promisesSettledResult[1] to be for the cluster deployment and
promisesSettledResult[0] will now be for the cluster deployment